### PR TITLE
Start session timeouts send fatal error and stop further executions

### PR DESCRIPTION
**Pull request by aggFTW**

### CODE REVIEW

**Comment by msftristew** [comment]
Doesn't have to happen in this PR, but we should really get the ball rolling on removing this "private magics" (the ones that begin with `_do_not_call`)

File: remotespark/kernels/kernelmagics.py [hunk 1]
```diff
@@ -30,6 +30,8 @@ def __init__(self, shell, data=None):
         self.url = None
         self.connection_string = None
```
Comment (MohamedElKamhawy) [inline remotespark/kernels/kernelmagics.py]: Remove extra line.

File: remotespark/kernels/kernelmagics.py [hunk 2]
```diff
@@ -190,7 +195,7 @@ def cleanup(self, line, cell="", local_ns=None):
             self.ipython_display.send_error("When you clean up the endpoint, all sessions will be lost, including the "
```
Comment (MohamedElKamhawy) [inline remotespark/kernels/kernelmagics.py]: This is not related to this PR. It's better to set a string variable with this sentence then print it.

File: remotespark/kernels/kernelmagics.py [hunk 3]
```diff
@@ -206,23 +211,38 @@ def delete(self, line, cell="", local_ns=None):
                 self.ipython_display.send_error("Cannot delete this kernel's session ({}). Specify a different session,"
```
Comment (MohamedElKamhawy) [inline remotespark/kernels/kernelmagics.py]: The same applied here.

File: remotespark/kernels/kernelmagics.py [hunk 4]
```diff
@@ -206,23 +211,38 @@ def delete(self, line, cell="", local_ns=None):
...
+
+        if not self.session_started:
             skip = False
             properties = conf.get_session_properties(self.language)
 
-            self.spark_controller.add_session(self.session_name, self.connection_string, skip, properties)
+            try:
+                self.session_started = True
+                self.spark_controller.add_session(self.session_name, self.connection_string, skip, properties)
+            except Exception as e:
```
Comment (MohamedElKamhawy) [inline remotespark/kernels/kernelmagics.py]: Leave an extra line before this line for readability.

File: tests/test_kernel_magics.py [hunk 5]
```diff
@@ -5,6 +5,7 @@
 import remotespark.utils.configuration as conf
 from remotespark.kernels.kernelmagics import KernelMagics
 from remotespark.utils.constants import Constants
+from remotespark.livyclientlib.livyclienttimeouterror import LivyClientTimeoutError
```
Comment (MohamedElKamhawy) [inline tests/test_kernel_magics.py]: Re-arrange alphabetically. They should be:
 from remotespark.kernels.kernelmagics import KernelMagics
 from remotespark.livyclientlib.livyclienttimeouterror import LivyClientTimeoutError
 from remotespark.utils.constants import Constants
import remotespark.utils.configuration as conf

**Comment by msftristew** [comment]
##APPROVED

File: tests/test_kernel_magics.py [hunk 6]
```diff
@@ -53,19 +54,42 @@ def test_start_session():
     line = ""
     assert not magic.session_started
 
-    magic._do_not_call_start_session(line)
+    ret = magic._do_not_call_start_session(line)
```
Comment (MohamedElKamhawy) [inline tests/test_kernel_magics.py]: Change the variable name to be more meaningful.

File: remotespark/kernels/kernelmagics.py (hunk 2 above)
Comment (aggFTW) [inline remotespark/kernels/kernelmagics.py]: ?

File: remotespark/kernels/kernelmagics.py (hunk 4 above)
Comment (aggFTW) [inline remotespark/kernels/kernelmagics.py]: That's not PEP

File: tests/test_kernel_magics.py (hunk 6 above)
Comment (aggFTW) [inline tests/test_kernel_magics.py]: Meaningful enough. It's a unit test.

Comment (MohamedElKamhawy) [inline tests/test_kernel_magics.py]: What "ret" stands for?

Comment (aggFTW) [inline tests/test_kernel_magics.py]: It's short for whatever is returned. Another very popular option is `ret_v`. Usually used in places where it's not really that meaningful, like a unit test.

File: remotespark/kernels/kernelmagics.py (hunk 4 above)
Comment (MohamedElKamhawy) [inline remotespark/kernels/kernelmagics.py]: What's "PEP"? Generally, this will help the code reader to identify whether this is a different statement separated from the try or not.

File: remotespark/kernels/kernelmagics.py (hunk 2 above)
Comment (MohamedElKamhawy) [inline remotespark/kernels/kernelmagics.py]: To use a string and doing whatever you want to do with it, it's better to save the string first in a new variable then use it.

File: remotespark/kernels/kernelmagics.py (hunk 4 above)
Comment (aggFTW) [inline remotespark/kernels/kernelmagics.py]: Did you search for it? That's what indentation is for.

File: remotespark/kernels/kernelmagics.py (hunk 2 above)
Comment (aggFTW) [inline remotespark/kernels/kernelmagics.py]: Maybe. I wouldn't say that as a blank statement.

File: tests/test_kernel_magics.py (hunk 6 above)
Comment (MohamedElKamhawy) [inline tests/test_kernel_magics.py]: Usually I prefer to have a meaningful variable name even if it's in unit test, this will help any new person to this code section to get the intent of the test easily.

File: remotespark/kernels/kernelmagics.py (hunk 4 above)
Comment (MohamedElKamhawy) [inline remotespark/kernels/kernelmagics.py]: Ok, that works.
